docs/src/preprocessing/preprocessing.md

```diff
@@ -1,6 +1,224 @@
+
+(3):
+- Bounding box $T$ is a leaf. Use open surface $T.\mathcal{S}$
+
+$w_{\text{right}} = \text{\texttt{naive\_winding}} (\mathbf{p}, T.\mathcal{S})$
+
+The reconstructed surface will then look as in the following image.
+
+```@raw html
+<figure>
+  <img src="https://github.com/user-attachments/assets/920bb4f1-1336-4e77-b06d-d5b46ca0d8d5" alt="reconstructed surface"/>
+</figure>
+```
+
+We finally sum up the winding numbers
+
+$w = w_{\text{left}} + w_{\text{right} } = -w_{T_{\text{left}}.\bar{\mathcal{S}}} + w_{T_{\text{right}}.\mathcal{S}}$
+
+#### Discrete geometry
+
+```@raw html
+<figure>
+  <img src="https://github.com/user-attachments/assets/a9b59cc3-5421-40af-b0b0-f4c18a5a7078" alt="discrete geometry"/>
+</figure>
+```
+
+To construct the hierarchy for discrete piecewise-linear surfaces (1), we have to do the following.
+
+(2):
+Each edge is distributed to the child whose box contains the edge's barycenter.
+Splitting stops when the number of a box's edges slips below a
+threshold (usually $\approx 100$ faces in 3D, here: 6 edges).
+
+(3):
+For the closure, Jacobson et al. (2013) define *exterior vertices* (*exterior edges* in 3D)
+as boundary vertices of such a segmentation.
+To find them, we traverse around each edge (face in 3D) in order, and
+increment or decrement for each vertex (edge) a specific counter.
+
+```julia
+  # 2D
+  v1 = edge_vertices_ids[edge][1]
+  v2 = edge_vertices_ids[edge][2]
+
+  vertex_count[v1] += 1
+  vertex_count[v2] -= 1
+```
+```julia
+  # 3D
+  edge1 = face_edges_ids[face][1]
+
+  if edge_vertices_ids[edge1] == (v1, v2)
+      directed_edges[edge1] += 1
+  else
+      directed_edges[edge1] -= 1
+  end
+```
+
+In 2D, a vertex is declared as exterior if `vertex_count(vertex) != 0`.
+The exterior vertices are then connected to one arbitrary
+exterior vertex using appropriately oriented line segments
+
+```julia
+  edge = vertex_count[v] > 0 ? (closing_vertex, v) : (v, closing_vertex)
+```
+
+In 3D, an edge is declared as exterior if `directed_edges[edge] != 0`
+The exterior edges are triangulated with one exterior vertex $k$ with orientation
+- ``\{i, j, k\}`` if `directed_edges[edge] < 0`
+- ``\{j, i, k\}`` if `directed_edges[edge] > 0`
+
+#### Incorrect evaluation
+
+If we follow the algorithm, we know that recursion stops if
+
+- The bounding box $T$ is a leaf
+- The query point $\mathbf{p}$ is outside the box
+
+(1): If $\mathbf{p}$ is outside the box, we calculate the winding number with the (red) closure of the box.
+
+(2): If $\mathbf{p}$ is inside the box, we use the (blue) edges distributed to the box.
+
+(3): In this case, it leads to an incorrect evaluation of the winding number.
+
+```@raw html
+<figure>
+  <img src="https://github.com/user-attachments/assets/7bae164a-8d5b-4761-9d54-9abf99fca94a" alt="incorrect evaluation"/>
+</figure>
+```
+
+#### Correct evaluation
+Jacobson et al. (2013) therefore resize the bounding box, since it doesn't matter if the boxes overlap.
```

Comment:
```suggestion
Jacobson et al. (2013) therefore resize the bounding box to fully include the closing surface of the neighboring box, since it doesn't matter if the boxes overlap.
```